clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -3027,6 +3037,13 @@ static void RenderFloatingPointOptions(const ToolChain &TC, const Driver &D,
               : ComplexArithmeticStr(LangOptions::ComplexRangeKind::CX_Basic));
     Range = LangOptions::ComplexRangeKind::CX_Basic;
     SeenUnsafeMathModeOption = true;
+    if (JA.isDeviceOffloading(Action::OFK_SYCL) ||
+        JA.isDeviceOffloading(Action::OFK_OpenMP)) {
```

Comment:
Can we use a boolean at the top of the function that covers?
`bool IsDeviceOffloading = JA.isDeviceOffloading(Action::OFK_SYCL);`
When we pulldown these to xmain, we can add the `OFK_OpenMP` check as part of this.